tests/components/lunatone/conftest.py

```diff
@@ -0,0 +1,105 @@
+"""Fixtures for Lunatone tests."""
+
+from collections.abc import Generator
+from unittest.mock import AsyncMock, PropertyMock, patch
+
+from lunatone_rest_api_client import Devices
+import pytest
+
+from homeassistant.components.lunatone.const import DOMAIN
+from homeassistant.const import CONF_URL
+
+from . import BASE_URL, DEVICES_DATA, INFO_DATA, SERIAL_NUMBER
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+def mock_setup_entry() -> Generator[AsyncMock]:
+    """Override async_setup_entry."""
+    with patch(
+        "homeassistant.components.lunatone.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        yield mock_setup_entry
+
+
+@pytest.fixture
+def mock_lunatone_auth() -> Generator[AsyncMock]:
+    """Mock a Lunatone auth object."""
+    with (
+        patch(
+            "homeassistant.components.lunatone.Auth",
+            autospec=True,
+        ) as mock_auth,
+        patch(
+            "homeassistant.components.lunatone.config_flow.Auth",
+            new=mock_auth,
+        ),
+    ):
+        auth = mock_auth.return_value
+        auth.base_url = BASE_URL
+        yield auth
+
+
+@pytest.fixture
+def mock_lunatone_devices() -> Generator[AsyncMock]:
+    """Mock a Lunatone devices object."""
+
+    def build_device_mock(devices: Devices):
+        device_list = []
+        for device_data in devices.data.devices:
+            device = AsyncMock(autospec=True)
+            device._data = device_data
+            type(device).data = PropertyMock(side_effect=lambda d=device: d._data)
+            device.id = device.data.id
+            device.name = device.data.name
+            type(device).is_on = PropertyMock(
+                side_effect=lambda d=device: d.data.features.switchable.status
+            )
```

Comment:
```suggestion
            device.data = device_data
            device.id = device.data.id
            device.name = device.data.name
            device.is_on = device.data.features.switchable.status
```
we can simplify this. also, lets only mock external methods, since internal ones are not used by the integration (this also requires changes in the tests, see my other comments)

Reply:
[8d491e1](https://github.com/home-assistant/core/pull/149182/commits/8d491e1c1a2af3b759138cdbc21e96fa0c2c8143)